src/clarity-angular/utils/conditional/if-active.directive.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    Output,
+    EventEmitter,
+    Inject,
+    OnDestroy
+} from "@angular/core";
+
+import { IfActiveService } from "./if-active.service";
+import {IF_ACTIVE_ID} from "../../layout/tabs/tab";
+import {Subject} from "rxjs/Subject";
+import "rxjs/add/operator/takeUntil";
+
+@Directive({
+    selector: "[clrIfActive]"
+})
+
+/**********
+ *
+ * @class IfActiveDirective
+ *
+ * @description
+ * A structural directive that controls whether or not the associated TemplateRef is instantiated or not.
+ * It makes use of a Component instance level service: IfActiveService to maintain state between itself and
+ * the component using it in the component template.
+ *
+ */
+export class IfActiveDirective implements OnDestroy {
+
+    private ngUnsubscribe: Subject<void> = new Subject<void>();
+
+    constructor( private ifActiveService: IfActiveService, @Inject(IF_ACTIVE_ID) private self: any,
```

Comment:
in `tab-content.ts` and `tab-link.ts`, `id` was already taken, so I was trying to use an alternative name (hence the `self`) and that got carried over to `if-active-directive.ts`. I'll rename the id's in `tab-content.ts` and `tab-link.ts` to `tabContentId` and `tabLinkId` and use `id` in every place for the `IF_ACTIVE_ID`.